utils/obsproc/NetCDFToIodaConverter.h

```diff
@@ -14,46 +14,106 @@
 #include "oops/util/DateTime.h"
 #include "oops/util/Duration.h"
 #include "oops/util/Logger.h"
+#include "oops/util/missingValues.h"
 
 namespace gdasapp {
+  // A simple data structure to organize the info to provide to the ioda
+  // writter
+  struct IodaVars {
+    int location;
+    int nVars;
+    Eigen::ArrayXf* obsVal;
+    Eigen::ArrayXf* obsError;
+    Eigen::ArrayXi* preQc;
+    std::string units;  // reference date for epoch time
+  };
+
+  // Base class for the converters
   class NetCDFToIodaConverter {
    public:
-    // Constructor that stores the configuration as a data member
-  explicit NetCDFToIodaConverter(const eckit::Configuration & fullConfig) {
-    // time window info
-    std::string winbegin;
-    std::string winend;
-    fullConfig.get("window begin", winbegin);
-    fullConfig.get("window end", winend);
-    this->windowBegin_ = util::DateTime(winbegin);
-    this->windowEnd_ = util::DateTime(winend);
-
-    // get input netcdf files
-    fullConfig.get("input files", this->inputFilenames_);
-
-    // ioda output file name
-    this->outputFilename_ = fullConfig.getString("output file");
-  }
-
-    // Method to write out a IODA file
+    // Constructor: Stores the configuration as a data members
+    explicit NetCDFToIodaConverter(const eckit::Configuration & fullConfig) {
+      // time window info
+      std::string winbegin;
+      std::string winend;
+      std::string obsvar;
+      fullConfig.get("window begin", winbegin);
+      fullConfig.get("window end", winend);
+      fullConfig.get("variable", obsvar);
+      this->windowBegin_ = util::DateTime(winbegin);
+      this->windowEnd_ = util::DateTime(winend);
+      this->variable_ = obsvar;
```

Comment:
usage of `this->` is not neccessary. Just do `windowBegin_ = util::DateTime(winbegin);`